wp-cache-phase2.php

```diff
@@ -1297,7 +1297,9 @@ function wp_cache_get_postid_from_comment( $comment_id, $status = 'NA' ) {
 	if ($postid > 0)  {
 		wp_cache_debug( "Post $postid changed. Update cache.", 4 );
 		return wp_cache_post_change( $postid );
-	} elseif ( $_GET[ 'delete_all' ] != 'Empty Trash' && $_GET[ 'delete_all2' ] != 'Empty Spam' ) {
+	} elseif ( isset( $GLOBALS[ 'pagenow' ] ) && $GLOBALS[ 'pagenow' ] === 'edit-comments.php' &&
```

Comment:
It isn't good. I'll try to check more details. Probably we need something like this :

`if ( isset( $GLOBALS[ 'pagenow' ] ) && $GLOBALS[ 'pagenow' ] === 'edit-comments.php' && ( isset( $_REQUEST['delete_all'] ) || isset( $_REQUEST['delete_all2'] ) ) ){  wp_cache_debug( "Empty trash or empty spam detected. Not deleting cache files.." ) }`